File: typescript/packages/legacy/x402/src/types/shared/network.ts
import { z } from "zod";

export const NetworkSchema = z.enum([
  "abstract",
  "abstract-testnet",
  "base-sepolia",
  "base",
  "avalanche-fuji",
  "avalanche",
  "iotex",
  "solana-devnet",
  "solana",
  "sei",
  "sei-testnet",
  "polygon",
  "polygon-amoy",
  "peaq",
  "story",
  "educhain",
  "skale-base-sepolia",
]);
export type Network = z.infer<typeof NetworkSchema>;

// evm
export const SupportedEVMNetworks: Network[] = [
  "abstract",
  "abstract-testnet",
  "base-sepolia",
  "base",
  "avalanche-fuji",
  "avalanche",
  "iotex",
  "sei",
  "sei-testnet",
  "polygon",
  "polygon-amoy",
  "peaq",
  "story",
  "educhain",
  "skale-base-sepolia",
];
export const EvmNetworkToChainId = new Map<Network, number>([
  ["abstract", 2741],
  ["abstract-testnet", 11124],
  ["base-sepolia", 84532],
  ["base", 8453],
  ["avalanche-fuji", 43113],
  ["avalanche", 43114],
  ["iotex", 4689],
  ["sei", 1329],
  ["sei-testnet", 1328],
  ["polygon", 137],
  ["polygon-amoy", 80002],
  ["peaq", 3338],
  ["story", 1514],
  ["educhain", 41923],
  ["skale-base-sepolia", 324705682],
]);

// svm
export const SupportedSVMNetworks: Network[] = ["solana-devnet", "solana"];
export const SvmNetworkToChainId = new Map<Network, number>([
  ["solana-devnet", 103],
  ["solana", 101],
]);

export const ChainIdToNetwork = Object.fromEntries(
  [...SupportedEVMNetworks, ...SupportedSVMNetworks].map(network => [
    EvmNetworkToChainId.get(network),
    network,
  ]),
) as Record<number, Network>;

File: go/mechanisms/evm/v1/network.go
package v1

import (
	"math/big"
)

// NetworkChainIDs maps v1 legacy network names to their chain IDs.
var NetworkChainIDs = map[string]*big.Int{
	"ethereum":           big.NewInt(1),
	"sepolia":            big.NewInt(11155111),
	"abstract":           big.NewInt(2741),
	"abstract-testnet":   big.NewInt(11124),
	"base-sepolia":       big.NewInt(84532),
	"base":               big.NewInt(8453),
	"avalanche-fuji":     big.NewInt(43113),
	"avalanche":          big.NewInt(43114),
	"iotex":              big.NewInt(4689),
	"sei":                big.NewInt(1329),
	"sei-testnet":        big.NewInt(1328),
	"polygon":            big.NewInt(137),
	"polygon-amoy":       big.NewInt(80002),
	"peaq":               big.NewInt(3338),
	"story":              big.NewInt(1514),
	"educhain":           big.NewInt(41923),
	"skale-base-sepolia": big.NewInt(324705682),
	"megaeth":            big.NewInt(4326),
	"monad":              big.NewInt(143),
	"stable":             big.NewInt(988),
	"stable-testnet":     big.NewInt(2201),
	"celo":               big.NewInt(42220),
	"flare":              big.NewInt(14),
}

// Networks is the list of all v1 network names.
var Networks []string

func init() {
	Networks = make([]string, 0, len(NetworkChainIDs))
	for name := range NetworkChainIDs {
		Networks = append(Networks, name)
	}
}
